orthoinference/src/main/java/org/reactome/orthoinference/EventsInferrer.java

```diff
@@ -165,19 +168,19 @@ public static void inferEvents(Properties props, String species) throws Exceptio
 			logger.info("Attempting RlE inference: " + reactionInst);
 			// Check if the current Reaction already exists for this species, that it is a valid instance (passes some filters), and that it doesn't have a Disease attribute.
 			// Adds to manualHumanEvents array if it passes conditions. This code block allows you to re-run the code without re-inferring instances.
-			List<GKInstance> previouslyInferredInstances = new ArrayList<GKInstance>();
+			List<GKInstance> previouslyInferredInstances = new ArrayList<>();
 			previouslyInferredInstances = checkIfPreviouslyInferred(reactionInst, orthologousEvent, previouslyInferredInstances);
-			previouslyInferredInstances = checkIfPreviouslyInferred(reactionInst, inferredFrom, previouslyInferredInstances);
-			if (previouslyInferredInstances.size() > 0)
-			{
+			previouslyInferredInstances.addAll(checkIfPreviouslyInferred(reactionInst, inferredFrom, previouslyInferredInstances));
```

Comment:
Since `previouslyInferredInstances` is only added to and not queried in the method `checkIfPreviouslyInferred` passing it as a parameter isn't needed.  It's best practice not to modify parameters, but instead to create and return a different variable in the method body.  

You could do this part of the code as follows:

```
List<GKInstance> previouslyInferredInstances = checkIfPreviouslyInferred(
    reactionInst, Arrays.asList(inferredFrom, orthologousEvent)
);
```

Then the `checkIfPreviouslyInferred` method could be:

```
private static List<GKInstance> checkIfPreviouslyInferred(GKInstance reactionInst, List<String> attributes) throws Exception 
{
	List previouslyInferredInstances = new ArrayList<>();
	for (String attribute : attributes) {
		for (GKInstance attributeInst : (Collection<GKInstance>) reactionInst.getAttributeValuesList(attribute)) {
			GKInstance reactionSpeciesInst = (GKInstance) attributeInst.getAttributeValue(species);
			if (reactionSpeciesInst.getDBID().equals(speciesInst.getDBID()) && attributeInst.getAttributeValue(isChimeric) == null) {
				previouslyInferredInstances.add(attributeInst);
			}
		}
	}
	return previouslyInferredInstances;
}
```

Reply:
Also, it may be better to rename `checkIfPreviouslyInferred` to `getInstancesIfPreviouslyInferred` to make it clear that the list of previously inferred instances is what the method returns.  A method name starting with "check" may be more suitable to a void or possibly a boolean return value.